api/scripts/arborescence-monitoring/time-series.js

```diff
@@ -0,0 +1,64 @@
+import { assertNotNullOrUndefined } from '../../src/shared/domain/models/asserts.js';
+
+const MAXIMUM_NUMBER_OF_POINTS_THRESHOLD = 28;
+
+const PURGE_POINT_INDEX = 10;
+
+const NB_POINTS_TO_PURGE = 1;
+
+const NULL_OR_UNDEFINED_POINTS_ERROR_MESSAGE = 'Time series should have a non null/undefined array of points';
+
+export class TimeSeries {
+  #points;
+  constructor(points) {
+    assertNotNullOrUndefined(points, NULL_OR_UNDEFINED_POINTS_ERROR_MESSAGE);
+    points.sort((pointA, pointB) => new Date(pointA.x).getTime() - new Date(pointB.x).getTime());
+    this.#points = points;
+  }
+
+  size() {
+    return this.#points.length;
```

Comment:
pinaillage pour sécuriser

```suggestion
    return this.#points..length ?? 0;
```

Reply:
Le constructeur nous assure que points est non null et pas undefined du coup je ne pense pas que les '??' soit nécessaire, non ?

Reply:
Dans ce cas il n'y a pas d'intérêt à cette fonction size, `this.#points.length` étant sécurisé et clair (et déjà utilisé dans cette classe).